modules/nextflow/src/main/groovy/nextflow/cli/CmdCid.groovy

```diff
@@ -165,14 +163,14 @@ class CmdCid extends CmdBase implements UsageAware{
         }
 
         private void printHistory(CidStore store) {
-            final historyFile = store.getHistoryFile()
-            if (historyFile.exists()) {
+            final records = store.historyLog?.records
+            if (records) {
                 def table = new TableBuilder(cellSeparator: '\t')
                     .head('TIMESTAMP')
                     .head('RUN NAME')
                     .head('SESSION ID')
                     .head('RUN CID')
-                historyFile.eachLine { table.append(CidHistoryFile.CidRecord.parse(it).toList()) }
+                records.forEach { table.append(it.toList()) }
```

Comment:
Nit, consider using plain `for` when possible to avoid nested classes (required by the closure)

Reply:
Modified